Modelica/Blocks/Sources.mo

```diff
@@ -340,10 +346,16 @@ The Real output y is a sine signal:
     annotation(Dialog(groupImage="modelica://Modelica/Resources/Images/Blocks/Sources/Cosine.png"));
     parameter SI.Frequency f(start=1) "Frequency of cosine wave";
     parameter SI.Angle phase=0 "Phase of cosine wave";
+    parameter Boolean continuous = false "If we want a smooth signal, the output will start at offset + amplitude*cos(phase)";
     extends Interfaces.SignalSource;
   equation
-    y = offset + (if time < startTime then 0 else amplitude*Modelica.Math.cos(2
-      *pi*f*(time - startTime) + phase));
+    if continuous then
+      y = offset + smooth(1, amplitude*(if time < startTime then Modelica.Math.cos(phase)
```

Comment:
Sorry, I was wrong about the cosine due to the shift with `phase`.  This also looks like `smooth(0, …)` to me.

Reply:
Looks good now.  Resolving.